### ISSUE TITLE
labwc crash

### ISSUE THREAD

**Issue by Reiddragon** [labwc/labwc#2371]
### Problem description

labwc crashed randomly after drag and drop

<details><summary>stack trace</summary>

```
           PID: 1373 (labwc)
           UID: 1000 (reid)
           GID: 1000 (reid)
        Signal: 11 (SEGV)
     Timestamp: Thu 2024-11-21 23:13:57 EET (8min ago)
  Command Line: labwc -C /home/reid/.config/labwc -S lxqt-session
    Executable: /usr/bin/labwc
 Control Group: /user.slice/user-1000.slice/session-1.scope
          Unit: session-1.scope
         Slice: user-1000.slice
       Session: 1
     Owner UID: 1000 (reid)
       Boot ID: 2825f5d4f7694476b04853ff10faa400
    Machine ID: 3c4eccf2d7694b658eb2742d0dfb9c1b
      Hostname: snow-raptor
       Storage: /var/lib/systemd/coredump/core.labwc.1000.2825f5d4f7694476b04853ff10faa400.1373.1732223637000000.zst (present)
  Size on Disk: 30M
       Message: Process 1373 (labwc) of user 1000 dumped core.
                
                Stack trace of thread 1373:
                #0  0x00007cc42b7f3078 wl_list_insert (libwayland-server.so.0 + 0x8078)
                #1  0x00007cc42b7f3450 wl_signal_emit_mutable (libwayland-server.so.0 + 0x8450)
                #2  0x00007cc42b72ec50 wlr_data_source_destroy (libwlroots-0.18.so + 0x59c50)
                #3  0x00007cc42b732177 wlr_seat_start_drag (libwlroots-0.18.so + 0x5d177)
                #4  0x00007cc42b7f347e wl_signal_emit_mutable (libwayland-server.so.0 + 0x847e)
                #5  0x00007cc42b730fbc wlr_seat_request_start_drag (libwlroots-0.18.so + 0x5bfbc)
                #6  0x00007cc42af6a596 n/a (libffi.so.8 + 0x7596)
                #7  0x00007cc42af6700e n/a (libffi.so.8 + 0x400e)
                #8  0x00007cc42af69bd3 ffi_call (libffi.so.8 + 0x6bd3)
                #9  0x00007cc42b7f1e85 n/a (libwayland-server.so.0 + 0x6e85)
                #10 0x00007cc42b7f6d22 n/a (libwayland-server.so.0 + 0xbd22)
                #11 0x00007cc42b7f5112 wl_event_loop_dispatch (libwayland-server.so.0 + 0xa112)
                #12 0x00007cc42b7f71f7 wl_display_run (libwayland-server.so.0 + 0xc1f7)
                #13 0x00006036cd4777e7 main (labwc + 0x147e7)
                #14 0x00007cc42a834e08 n/a (libc.so.6 + 0x25e08)
                #15 0x00007cc42a834ecc __libc_start_main (libc.so.6 + 0x25ecc)
                #16 0x00006036cd477915 _start (labwc + 0x14915)

```

</details>

### Steps to reproduce

unknown

### labwc build source

Release

### labwc version

0.8.1

### labwc environment

From a TTY or some display manager like lightdm

### Distribution

Arch Linux

**Comment by Consolatis** [comment]
This has something to do with drag-and-drop, maybe between x11 and wayland. Potentially the same as
- #2287

**Comment by Reiddragon** [comment]
the drag-and-drop sounds quite likely, but I can say for certain I had no X11 windows running

**Comment by drigoskalwalker** [comment]
I got randons crashs on labwc using nouveau open driver, after changed to nvidia binaries the problem was gone.

**Comment by Consolatis** [comment]
I am pretty sure this is the same as https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3925 which should be fixed once wlroots 0.18.2 releases.

In the meantime, we have two options from the labwc side:
- force-enable xwaylandPersistence until we can depend on wlroots 0.18.2
- change the default of xwaylandPersistence to `true`

My personal preference would be to force-enable it as otherwise it is just a crash waiting to happen. Note that the dnd operation does not need to involve a X11 window at all (but if it does it makes it way more likely to happen due to another wlroots bug that hopefully will also be fixed by a 0.18.2 release).

**Comment by Consolatis** [comment]
Thoughts @ahesford @jlindgren90 @johanmalm @tokyo4j ?

**Comment by ahesford** [comment]
I'd rather just change the default. Anybody that has deliberately specified a false value for the persistence flag shouldn't have it ignored, and all other people will get the right behavior out of the box.

Because I never use drag and drop and almost never need X11, I like that the X server launches on demand and dies shortly thereafter. Keeping it running to avoid a bug I'll never hit seems counterproductive for me.

**Comment by johanmalm** [comment]
Same view as ahesford

**Comment by Consolatis** [comment]
Alright. So gameplan is then
- change default to persistent until we can depend on wlroots 0.18.2
- add note in NEWS.md warning about the potential crash

Does that sound sensible?

**Comment by johanmalm** [comment]
Yep.

Further consideration:

- Set `xwaylandPersistence=true` by default forever. Might be more confusing for users if we swap back and forth.

Do we know if this is a problem on the v0.7 branch. Or is it wlroots-0.18 related?

**Comment by Consolatis** [comment]
> * Set `xwaylandPersistence=true` by default forever.

I actually quite like the lazy approach as a default (in general if it doesn't trigger a crash).

> Might be more confusing for users if we swap back and forth.

I think as long as its communicated clearly in NEWS.md its not an issue. I also expect it to be just for one labwc release. Most users will likely not even notice a difference anyway because in both cases their X11 applications just work.

> Do we know if this is a problem on the v0.7 branch. Or is it wlroots-0.18 related?

I never looked into it but I assume it is an issue for as long as wlroots supports wayland -> xwayland drag and drop and xwayland lazy startup.

**Comment by johanmalm** [comment]
Okay. Agree. Let's got for a temporary change of default.

**Comment by tokyo4j** [comment]
In addition, I suggest showing an error message when `xwaylandPersistence` is set `no` like: `setting xwaylandPersistence to yes is not encouraged now since it has a potential risk of crashing the entire session. See #2371 for details`.

**Comment by Consolatis** [comment]
Should be fixed by using the just released wlroots 0.18.2. Please comment if you experience further crashes in relation to DND.
